include/awkward/array/EmptyArray.h

```diff
@@ -1,58 +1,77 @@
+  virtual void settype_part(const std::shared_ptr<Type> type);
+  virtual bool accepts(const std::shared_ptr<Type> type);
+  virtual int64_t length() const;
+  virtual const std::shared_ptr<Content> shallow_copy() const;
+  virtual void check_for_iteration() const;
+  virtual const std::shared_ptr<Content> getitem_nothing() const;
+  virtual const std::shared_ptr<Content> getitem_at(int64_t at) const;
+  virtual const std::shared_ptr<Content> getitem_at_nowrap(int64_t at) const;
+  virtual const std::shared_ptr<Content> getitem_range(int64_t start,
+                                                       int64_t stop) const;
+  virtual const std::shared_ptr<Content>
+  getitem_range_nowrap(int64_t start, int64_t stop) const;
+  virtual const std::shared_ptr<Content>
+  getitem_field(const std::string &key) const;
+  virtual const std::shared_ptr<Content>
+  getitem_fields(const std::vector<std::string> &keys) const;
+  virtual const std::shared_ptr<Content> carry(const Index64 &carry) const;
+  virtual const std::pair<int64_t, int64_t> minmax_depth() const;
+  virtual int64_t numfields() const;
+  virtual int64_t fieldindex(const std::string &key) const;
+  virtual const std::string key(int64_t fieldindex) const;
+  virtual bool haskey(const std::string &key) const;
+  virtual const std::vector<std::string> keyaliases(int64_t fieldindex) const;
+  virtual const std::vector<std::string>
+  keyaliases(const std::string &key) const;
+  virtual const std::vector<std::string> keys() const;
 
-  protected:
-    virtual const std::shared_ptr<Content> getitem_next(const SliceAt& at, const Slice& tail, const Index64& advanced) const;
-    virtual const std::shared_ptr<Content> getitem_next(const SliceRange& range, const Slice& tail, const Index64& advanced) const;
-    virtual const std::shared_ptr<Content> getitem_next(const SliceArray64& array, const Slice& tail, const Index64& advanced) const;
-    virtual const std::shared_ptr<Content> getitem_next(const SliceField& field, const Slice& tail, const Index64& advanced) const;
-    virtual const std::shared_ptr<Content> getitem_next(const SliceFields& fields, const Slice& tail, const Index64& advanced) const;
-  };
-}
+protected:
+  virtual const std::shared_ptr<Content>
+  getitem_next(const SliceAt &at, const Slice &tail,
+               const Index64 &advanced) const;
+  virtual const std::shared_ptr<Content>
+  getitem_next(const SliceRange &range, const Slice &tail,
+               const Index64 &advanced) const;
+  virtual const std::shared_ptr<Content>
+  getitem_next(const SliceArray64 &array, const Slice &tail,
+               const Index64 &advanced) const;
+  virtual const std::shared_ptr<Content>
+  getitem_next(const SliceField &field, const Slice &tail,
+               const Index64 &advanced) const;
+  virtual const std::shared_ptr<Content>
+  getitem_next(const SliceFields &fields, const Slice &tail,
+               const Index64 &advanced) const;
```

Comment:
The wide form of these was easier to read because it packs similar function names and arguments into something that can be vertically scanned. And this one goes wider than the GitHub window.

Okay, maybe no hard linewidth-boundary at all.

I thought this program was going to raise alerts about things it finds problematic and we can decide if we want to pay attention to those alerts on a case-by-case basis. If it automatically changes it, I don't think I like it.